### PR TITLE
[FTAMS102020] Emma Northridge

### PR DESCRIPTION
**Pull request by emmanorthridge**
Completed

### CODE REVIEW

File: src/viking.js [hunk 1]
```diff
@@ -1,11 +1,103 @@
...
+         this.vikingArmy = this.vikingArmy.filter(Viking => Viking.health > 0);
+         return vikingDamage
+     }
+
+     showStatus() {
+         if (this.saxonArmy.length === 0) {
+             return "Vikings have won the war of the century!" }
+             else if (this.vikingArmy.length === 0) {
+                 return "Saxons have fought for their lives and survived another day..." }
+                 else if (this.saxonArmy.length === this.vikingArmy.length) {
```
Comment (LloydC) [inline src/viking.js:92]: That's incorrect, the the right condition for Vikings and Saxons to still be at war is for their armies to exist
```
else if (this.vikingArmy.length > 0 && this.saxonArmy.length > 0) {
      return 'Vikings and Saxons are still in the thick of battle.';
    }

File: src/viking.js [hunk 2]
```diff
@@ -1,11 +1,103 @@
...
+    constructor(health, strength) {
+        super(health, strength)
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`;
+        }
+        if (this.health <= 0) {
```
Comment (LloydC) [inline src/viking.js:49]: You can also use `else{...}` here

File: src/viking.js [hunk 3]
```diff
@@ -1,11 +1,103 @@
...
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage; 
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`;
+        }
+        if (this.health <= 0) {
```
Comment (LloydC) [inline src/viking.js:27]: You can use `else{...}` as well

File: src/viking.js [hunk 4]
```diff
@@ -1,11 +1,103 @@
...
+
+    addSaxon(Saxon) {
+        this.saxonArmy.push(Saxon);
+    }
+
+    vikingAttack() {
+        const randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+        const randomSaxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+        const saxonDamage = randomSaxon.receiveDamage(randomViking.strength);
+        this.saxonArmy = this.saxonArmy.filter(Saxon => Saxon.health > 0);
```
Comment (LloydC) [inline src/viking.js:75]: To change the saxonArmy according to the damage, you can use this on line 75
```
if (randomSaxon.health <= 0) {
      this.saxonArmy.splice(randomSaxonIndex, 1);
    }

File: src/viking.js [hunk 5]
```diff
@@ -1,11 +1,103 @@
...
+        return saxonDamage;
+     }
+
+     saxonAttack() {
+         const randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+         const randomSaxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+         const vikingDamage = randomViking.receiveDamage(randomSaxon.strength);
+         this.vikingArmy = this.vikingArmy.filter(Viking => Viking.health > 0);
```
Comment (LloydC) [inline src/viking.js:83]: Same logic, you can use this on line 83
```
if (randomViking.health <= 0) {
      this.vikingArmy.splice(randomVikingIndex, 1);
    }

**Comment by emmanorthridge** [comment]
@LloydC @wax-bear 
Thanks for the constructive feedback!
I have updated my code.

**Pull request closed without merging**